main.go

```diff
@@ -779,6 +779,20 @@ func initialiseSystem(arguments map[string]interface{}) error {
 		afterConfSetup(&config.Global)
 	}
 
+	if os.Getenv("TYK_LOGLEVEL") == "" && arguments["--debug"] == false && config.Global.LogLevel != "" {
+		switch strings.ToLower(config.Global.LogLevel) {
+		case "error":
+			log.Level = logrus.ErrorLevel
+		case "warn":
+			log.Level = logrus.WarnLevel
+		case "debug":
+			log.Level = logrus.DebugLevel
+		}
```

Comment:
I would consider having a default in your switch case for unsupported log levels.

Reply:
From what I can tell, the default case is already handled when the logger is initialised by logger.get() when creating the log var where it will always default to info level.

Reply:
Josh is right. In `--debug`, we don't set it to `InfoLevel` if the flag isn't set, so this is consistent.

Reply:
However, you want to error if the config has something that is none of these, `""`, nor `"info"`. You could do:

```
switch strings.ToLower(config.Global.LogLevel) {
case "", "info":
	// default, do nothing
case "error":
	log.Level = logrus.ErrorLevel
case "warn":
	log.Level = logrus.WarnLevel
case "debug":
	log.Level = logrus.DebugLevel
default:
	log.WithFields(...).Fatal("invalid log level in the config bla bla bla")
 }
```

Otherwise, setting `"log_level": "verbose"` or `"log_level": "debgu"` and such could cause confusion.